crates/bevy_asset/src/io/file/file_watcher.rs

```diff
@@ -31,11 +33,12 @@ pub struct FileWatcher {
 impl FileWatcher {
     /// Creates a new [`FileWatcher`] that watches for changes to the asset files in the given `path`.
     pub fn new(
-        path: PathBuf,
+        path: String,
         sender: Sender<AssetSourceEvent>,
         debounce_wait_time: Duration,
     ) -> Result<Self, notify::Error> {
-        let root = normalize_path(&path).canonicalize()?;
```

Comment:
I believe this canonicalize is important here. It makes paths with dots work correctly `"../../my_assets_dir"`. Relevant PR: https://github.com/bevyengine/bevy/pull/18345 (hey look its me! I forgot all about this lol)

Should FileWatcher even be dealing with asset paths? This seems like a case where a file system path actually does make sense? I'm not sure though.

Reply:
I'm not familiar enough (yet?) with the codebase to decide, so I'd love some guidance here

Reply:
I looked into this a bit more, and I'm doubling down: I think FileWatcher should continue to deal with `Path` directly. I think the main thing we want to replace is the fact that our asset paths (as in, the ones that we pass into `asset_server.load()`) include some platform specific nonsense. That **does not** mean that our entire stack needs to deal only with strings - at some point, we need to convert that asset path into a concrete file path, so a `Path` instance.

So coming at it from the other angle, `FileWatcher` should be dealing with concrete paths because it is listening for real file system events for a particular directory. So I think we should revert what's happening in this `file_watcher` mod.